course/viewbase.py

```diff
@@ -67,11 +68,12 @@ def get_resource_objects(self):
             self.url_without_language = remove_query_param_from_url(self.request.get_full_path(), 'hl')
             self.query_language = None
             self.user_language = None
+            self.points = CachedPoints(self.instance, self.request.user, self.content, self.is_course_staff)
```

Comment:
@PasiSa this crashes in the interactions between A+ and Gitmanager.
`self.request.user` is a `GraderUser`, but `CachedPoints` does not work with `GraderUser`.

```
aplus-plus-1        | [2023-09-21 00:45:45,023: ERROR/log] Internal Server Error: /api/v2/courses/2/send_mail/
aplus-plus-1        | Traceback (most recent call last):
aplus-plus-1        |   File "/usr/local/lib/python3.9/dist-packages/rest_framework/views.py", line 497, in dispatch
aplus-plus-1        |     self.initial(request, *args, **kwargs)
aplus-plus-1        |   File "/srv/aplus/authorization/api/mixins.py", line 14, in initial
aplus-plus-1        |     self.get_resource_objects()
aplus-plus-1        |   File "/srv/aplus/course/viewbase.py", line 71, in get_resource_objects
aplus-plus-1        |     self.points = CachedPoints(self.instance, self.request.user, self.content, self.is_course_staff)
aplus-plus-1        |   File "/srv/aplus/exercise/cache/points.py", line 69, in __init__
aplus-plus-1        |     super().__init__(course_instance, user)
aplus-plus-1        |   File "/srv/aplus/lib/cache/cached.py", line 33, in __init__
aplus-plus-1        |     self.data = self.__get_data()
aplus-plus-1        |   File "/srv/aplus/lib/cache/cached.py", line 61, in __get_data
aplus-plus-1        |     data = self._generate_data(*self.__models, data=data)
aplus-plus-1        |   File "/srv/aplus/exercise/cache/points.py", line 91, in _generate_data
aplus-plus-1        |     user.userprofile.submissions
aplus-plus-1        | AttributeError: 'GraderUser' object has no attribute 'submissions'

```

Reply:
My workaround: https://github.com/apluslms/a-plus/commit/a8e76f7e7463c19be1eb31c04781f07f9de90682